builder/dockerfile/internals_windows.go

```diff
@@ -1,7 +1,101 @@
 package dockerfile
 
-import "github.com/docker/docker/pkg/idtools"
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
 
-func parseChownFlag(chown, ctrRootPath string, idMappings *idtools.IDMappings) (idtools.IDPair, error) {
-	return idMappings.RootPair(), nil
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/pkg/idtools"
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/docker/docker/pkg/system"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+func parseChownFlag(builder *Builder, state *dispatchState, chown, ctrRootPath string, identityMapping *idtools.IdentityMapping) (idtools.Identity, error) {
+	if builder.options.Platform == "windows" {
+		return getAccountIdentity(builder, chown, ctrRootPath, state)
+	}
+
+	return identityMapping.RootPair(), nil
+}
+
+func getAccountIdentity(builder *Builder, accountName string, ctrRootPath string, state *dispatchState) (idtools.Identity, error) {
+	sid, _, accType, err := windows.LookupSID("", accountName)
+
+	if err == nil && (accType == windows.SidTypeAlias || accType == windows.SidTypeWellKnownGroup) {
+		accountSid, err := sid.String()
+
+		if err != nil {
+			return idtools.Identity{SID: ""}, errors.Wrapf(err, "error converting SID to string")
+		}
+
+		return idtools.Identity{SID: accountSid}, nil
+	}
+
+	if strings.EqualFold(accountName, "ContainerAdministrator") {
+		return idtools.Identity{SID: "S-1-5-93-2-1"}, nil
+
+	} else if strings.EqualFold(accountName, "ContainerUser") {
+		return idtools.Identity{SID: "S-1-5-93-2-2"}, nil
+	}
+
+	accountSid, err := lookupNTAccount(builder, accountName, state)
+
+	identity := idtools.Identity{SID: accountSid}
+
+	return identity, err
+}
+
+func lookupNTAccount(builder *Builder, accountName string, state *dispatchState) (string, error) {
+
+	source, _ := filepath.Split(os.Args[0])
+
+	target := "C:\\Docker"
+	targetExecutable := target + "\\containerutility.exe"
```

Comment:
Guess we still need to discuss how we get this binary included in the builds (if we compile from source, or pull a released binary)

Reply:
Yes, although I don't think that should block the PR being merged.